File: public/javascripts/model/piece.js
define([], function() {
    function Piece(colour, location, board) {
        var colour  = colour; 
        var location = location;
        var board = board;
        var direction = 1;
        var moved = 0;
        var moves;

        if (colour == WHITE) {
            direction = 1;
        } else {
            direction = -1;
        }
    }

    return Piece;
});

File: app/model/Piece.java
package com.chess.app;

public interface Piece {
    Moves getMoves();

    void moveTo(Location dest);

    void beforeMove(Location src, Location dest);

    boolean isOpponent(Colour otherColour);

    boolean isWhite();

    boolean hasNotMoved();

    void destroy();

    void setLocation(Location location);
    
    Location getLocation();

    void setBoard(Board board);

    Player getOwner();

    void setOwner(Player player);

    Moves getCaptureMoves();

    boolean isKing();
}
